pkg/kv/kvserver/store_create_replica.go

```diff
@@ -25,10 +25,25 @@ import (
 
 var errRetry = errors.New("retry: orphaned replica")
 
-// getOrCreateReplica returns a replica for the given RangeID, creating an
-// uninitialized replica if necessary. The caller must not hold the store's
-// lock. The returned replica has Replica.raftMu locked and it is the caller's
-// responsibility to unlock it.
+// getOrCreateReplica returns a replica with the given replicaID for the given
+// rangeID, or roachpb.RaftGroupDeletedError if this replicaID has been deleted.
+// If the replica is not in memory, but also is not deleted, it creates and
+// returns an uninitialized replica. The returned replica's Replica.raftMu is
+// locked, and the caller is responsible for unlocking it.
+//
+// If getOrCreateReplica returns a replica, the guarantee is that the following
+// invariants are true while Replica.raftMu is held:
+//
+//   - Store.GetReplica(rangeID) successfully returns this and only this replica
+//   - The RangeTombstone in storage does not see this replica as removed
```

Comment:
nor the `r.mu.destroyStatus` field, I hope there is a comment on that field that it always mirrors the RangeTombstone  and is mutated only in a raftMu critical section. If not, could you add one?

Reply:
The comment [mentions](https://github.com/cockroachdb/cockroach/blob/15bc0c4780/pkg/kv/kvserver/replica.go#L330-L334) the locks, yes.

Re `RangeTombstone` mirroring, could you clarify what that means? Is the tombstone always written first, before `destroyStatus` is updated? Are there situations in which `destroyStatus` can be updated independently / before the tombstone? Wasn't apparent to me when looking at its usage.

Reply:
Ah, actually, looking at the tombstone usage helped. It's written only in one place, and only after asserting that `destroyStatus` is `Removed`.

Reply:
Added a comment.